### ISSUE TITLE
Null pointer exception in Jenkins build

### ISSUE THREAD

**Issue by Shyam-Gupta** [dotnet/roslyn#28407]
PR: https://github.com/dotnet/roslyn/pull/28406

Error in windows_debug_spanish_unit32_prtest

https://ci.dot.net/job/dotnet_roslyn/job/features_compiler/job/windows_debug_spanish_unit32_prtest/67/consoleFull#-410211996a086b3e-df04-41d2-bc4d-43e8f9406d07

23:35:25 ERROR: Build step failed with exception
23:35:25 java.lang.NullPointerException
23:35:25 at org.jenkinsci.plugins.xunit.XUnitProcessor.performTests(XUnitProcessor.java:145)
23:35:25 at org.jenkinsci.plugins.xunit.XUnitProcessor.performXUnit(XUnitProcessor.java:88)
23:35:25 at org.jenkinsci.plugins.xunit.XUnitPublisher.perform(XUnitPublisher.java:142)
23:35:25 at org.jenkinsci.plugins.xunit.XUnitPublisher.perform(XUnitPublisher.java:134)
23:35:25 at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
23:35:25 at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:744)
23:35:25 at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:690)
23:35:25 at hudson.model.Build$BuildExecution.post2(Build.java:186)
23:35:25 at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:635)
23:35:25 at hudson.model.Run.execute(Run.java:1749)
23:35:25 at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
23:35:25 at hudson.model.ResourceController.execute(ResourceController.java:97)
23:35:25 at hudson.model.Executor.run(Executor.java:429)

**Comment by dpoeschl** [comment]
There's an error slightly before that:
```
00:20:11.424 FATAL: command execution failed
00:20:11.424 java.nio.channels.ClosedChannelException
00:20:11.424 	at org.jenkinsci.remoting.protocol.NetworkLayer.onRecvClosed(NetworkLayer.java:154)
00:20:11.424 	at org.jenkinsci.remoting.protocol.impl.NIONetworkLayer.ready(NIONetworkLayer.java:179)
00:20:11.424 	at org.jenkinsci.remoting.protocol.IOHub$OnReady.run(IOHub.java:721)
```

**Comment by dpoeschl** [comment]
Another one: https://ci.dot.net/job/dotnet_roslyn/job/dev15.8-preview4-vs-deps/job/windows_release_vs-integration_prtest/15/consoleFull#-410211996a086b3e-df04-41d2-bc4d-43e8f9406d07

**Comment by tannergooding** [comment]
FYI. @dotnet/dnceng

**Comment by MattGal** [comment]
Filed https://github.com/dotnet/core-eng/issues/3861 to track this, I think this is a side effect of which Tuesday of the month this is sadly.

**Comment by dpoeschl** [comment]
Another one from https://github.com/dotnet/roslyn/pull/28442: https://ci.dot.net/job/dotnet_roslyn/job/dev15.8-preview4-vs-deps/job/windows_debug_vs-integration_prtest/16/consoleFull#9373088216ee26338-6221-4f7c-8c8f-1d1d5cb4704a in

**Comment by MattGal** [comment]
@dpoeschl we continue to have machines rebooting during runs due to Windows update.  I believe I've addressed this specific one, please continue to add them here or  dotnet/core-eng#3861 if you see more.

**Comment by MattGal** [comment]
I was unable to find any repro'ing from surprise reboots and the fix made for dotnet/core-eng#3861 is definitely on new machines; I'm closing that issue, please ping me if you see more of these and we can investigate again.
